### PR TITLE
AK+FileManager: Add option to string formatter to use a digit separator (and use it in PropertiesWindow)

### PR DESCRIPTION
**Pull request by tcl3**
With this PR, `vformat()` can now accept format specifiers of the form  `{:'[numeric-type]}`. This will output a number with a comma separator every 3 digits.

For example:

`dbgln("{:'d}", 9999999);` will output 9,999,999.

Binary, octal and hexadecimal numbers can also use this feature, for
example:

`dbgln("{:'x}", 0xffffffff);` will output ff,fff,fff.

The type specifier can also be omitted, so:

`dbgln("{:'}", 9999999);` will output 9,999,999

I used the apostrophe for this option, as that is what `printf()` uses. I'm not a huge fan though, so I'm open to suggestions if there is something better.

As part of this PR I've updated the FileManager's PropertiesWindow to use the new option.

propertieswindow_updated

### CODE REVIEW

**Comment by tcl3** [comment]
I didn't notice that someone else had already submitted #17719. I'm going to close this PR in favour of that one.

**Comment by MacDue** [comment]
I like the integration with `vformat()`, I feel like that could still be merged.

**Comment by tcl3** [comment]
I have reopened the PR; I believe the `vformat()` functionality could still be useful. I have currently set it to draft, as I am not sure how to proceed with it. 

I'm not sure whether it would be best to remove the FileManager related commits from the PR in favor of #17719. I don't think there is anything wrong with the other implementation, although I'm not sure how necessary it is to have multiple ways to achieve the same thing.

I've had a look at how this feature is implemented in other places. Here's a comparison to the implementation in this PR:

```c++
setlocale( LC_NUMERIC, "en_US.UTF-8");
printf("%'\n", 16777216); // Prints: %'
printf("%'d\n", 16777216); // Prints: 16,777,216
printf("%'x\n", 16777216); // Prints: 1,000,000
   
std::locale::global(std::locale("en_US.UTF-8"));
std::cout << std::format("{:L}", 16777216) << std::endl; // Prints: 16,777,216
std::cout << std::format("{:Ld}", 16777216) << std::endl; // Prints: 16,777,216
std::cout << std::format("{:Lx}", 16777216) << std::endl; // Prints: 1,000,000
   
dbgln("{:'}", 16777216); // Prints 16,777,216
dbgln("{:'d}", 16777216); // Prints 16,777,216
dbgln("{:'x}", 16777216); // Prints: 100_0000   
```

Given the above, I don't know whether it's worth keeping the functionality that differentiates between base 10 and non base 10 numbers, as it isn't really based on anything, other than the fact I thought it looked slightly better.

**Comment by tcl3** [comment]
My latest push removes the logic which changes the separator depending on the base. I think the benefits of having this are questionable, and this sort of differentiation doesn't happen in other libraries. 

I've updated the PR description and the example above to reflect this change.

File: AK/NumberFormat.cpp [hunk 1]
```diff
@@ -56,12 +56,15 @@ DeprecatedString human_readable_size(u64 size, HumanReadableBasedOn based_on)
     return human_readable_quantity(size, based_on, "B"sv);
 }
 
-DeprecatedString human_readable_size_long(u64 size)
+DeprecatedString human_readable_size_long(u64 size, UseThousandsSeparator use_thousands_separator)
 {
     if (size < 1 * KiB)
         return DeprecatedString::formatted("{} bytes", size);
```
Comment (GMTA) [inline AK/NumberFormat.cpp:62]: If `size == 1023`, the thousands separator wouldn't show up here.

**Comment by stelar7** [comment]
Could this be updated to allow different separator chars?
In my locale `.`, `'` and ` ` are used a lot more than `,`

**Comment by tcl3** [comment]
My latest push updates `NumberFormat::human_readable_size()`, so that numbers between 1000 and 1023 (of any unit) are displayed with a separator if `UseThousandsSeperator` is set to `UseThousandsSeperator::Yes` . I have updated the screenshot above to reflect this change.

**Pull request merged**